src/Dfc.ProviderPortal.ChangeFeedListener/Services/CourseAuditService.cs

```diff
@@ -153,12 +152,23 @@ public async Task RepopulateSearchIndex(ILogger log)
                     new FeedOptions() { EnableCrossPartitionQuery = true })
                     .AsDocumentQuery();
 
+                var indexedIds = new List<string>();
+
                 while (query.HasMoreResults)
                 {
                     var result = await query.ExecuteNextAsync<Document>();
 
-                    var indexed = await searchServiceWrapper.UploadBatch(allProviders, allVenues, result);
-                    indexedCount += indexed.Count();
+                    var indexed = await searchServiceWrapper.UploadBatch(updateBatchId, allProviders, allVenues, result);
+                    indexedIds.AddRange(indexed);
+                    indexedCount += indexed.Count;
+                }
+
+                if (deleteZombies)
+                {
+                    // Any records that haven't just been updated are zombies - go forth and reap
```

Comment:
```suggestion
                    // Any records that haven't just been updated should not be in the search index because they don't exist in the source data. Delete them.
```